bug(di): `#[injectable_factory]` generates orphan `Injectable` impl for external return types

## Description

PR #3435 added automatic `Injectable` impl generation in `#[injectable_factory]`. However, when the factory's return type is defined in an external crate (e.g., `CookieSessionConfig`, `UnifiedRouter` from reinhardt), the generated impl violates Rust's orphan rule (E0117).

## Steps to Reproduce

```rust
use reinhardt::CookieSessionConfig;
use reinhardt::di::injectable_factory;

// CookieSessionConfig is defined in reinhardt, not in the current crate
#[injectable_factory(scope = "singleton")]
async fn create_cookie_session_config() -> CookieSessionConfig {
    CookieSessionConfig { /* ... */ }
}
```

## Error

```
error[E0117]: only traits defined in the current crate can be implemented for types defined outside of the crate
  --> dashboard/src/config/urls.rs:66:1
   |
66 | #[injectable_factory(scope = "singleton")]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

## Root Cause

`#[injectable_factory]` unconditionally generates:

```rust
#[async_trait]
impl Injectable for CookieSessionConfig {
    async fn inject(ctx: &InjectionContext) -> DiResult<Self> {
        let arc = ctx.__resolve_from_registry::<Self>().await?;
        Ok((*arc).clone())
    }
}
```

This violates E0117 because both `Injectable` (from `reinhardt-di`) and `CookieSessionConfig` (from `reinhardt-middleware`) are external to the downstream crate.

## Suggested Fix

**Option A (recommended)**: Skip `Injectable` impl generation when the return type is likely external. The macro can't determine this at compile time, so add an opt-out attribute:

```rust
#[injectable_factory(scope = "singleton", no_injectable_impl)]
async fn create_cookie_session_config() -> CookieSessionConfig { ... }
```

**Option B**: Generate the `Injectable` impl inside the reinhardt crate itself for common types (`CookieSessionConfig`, `UnifiedRouter`, etc.) using a blanket impl that delegates to the registry.

**Option C**: Provide a blanket `Injectable` impl for all types registered via `#[injectable_factory]` using a marker trait. This would require the registry to be consulted at the trait level.

## Additional Issues

The generated `Injectable` impl also requires `Clone` on the return type. For types like `AllowedOrigins` (local newtype), adding `#[derive(Clone)]` is straightforward. But for external types, `Clone` may or may not be implemented.

Also, the generated code references `reinhardt_di::async_trait` which may not be re-exported:

```
error[E0433]: failed to resolve: could not find `async_trait` in `reinhardt_di`
```

## Discovered In

reinhardt-cloud `feat/cookie-session-auth` branch, after updating reinhardt-web to `3695e446a` (includes PR #3435).

--- a/crates/reinhardt-admin/src/core/database.rs
+++ b/crates/reinhardt-admin/src/core/database.rs
@@ -1276,6 +1276,22 @@ impl Injectable for AdminDatabase {
 	}
 }
 
+// Register AdminDatabase in the global dependency registry so that
+// Depends<AdminDatabase> can resolve it via ctx.resolve().
+// Delegates to Injectable::inject() for lazy construction from DatabaseConnection.
+fn __register_admin_database(registry: &reinhardt_di::DependencyRegistry) {
+	registry.register::<AdminDatabase>(
+		reinhardt_di::DependencyScope::Singleton,
+		reinhardt_di::InjectableFactory::<AdminDatabase>::new(),
+	);
+}
+
+reinhardt_di::inventory::submit! {
+	reinhardt_di::InjectableRegistration::new(
+		__register_admin_database
+	)
+}
+
 #[cfg(test)]
 mod tests {
 	use super::*;

--- a/crates/reinhardt-di/macros/src/injectable_factory.rs
+++ b/crates/reinhardt-di/macros/src/injectable_factory.rs
@@ -193,17 +193,6 @@ pub(crate) fn injectable_factory_impl(args: TokenStream, input: ItemFn) -> Resul
 			)
 		}
 
-		// Bridge Injectable trait to registry-based resolution
-		// so that Depends<#return_type> works for factory-registered types.
-		// Uses __resolve_from_registry to bypass cycle detection (the caller,
-		// Depends::resolve, has already registered this type in the cycle stack).
-		#[#di_crate::async_trait::async_trait]
-		impl #di_crate::Injectable for #return_type {
-			async fn inject(ctx: &#di_crate::InjectionContext) -> #di_crate::DiResult<Self> {
-				let __arc = ctx.__resolve_from_registry::<Self>().await?;
-				Ok((*__arc).clone())
-			}
-		}
 	};
 
 	Ok(expanded)

--- a/crates/reinhardt-di/src/context.rs
+++ b/crates/reinhardt-di/src/context.rs
@@ -618,11 +618,23 @@ impl InjectionContext {
 			register_type_name::<T>(type_name);
 
 			// [Fast path] Skip circular detection on cache hit
-			let scope = registry.get_scope::<T>().ok_or_else(|| {
-				crate::DiError::DependencyNotRegistered {
-					type_name: type_name.to_string(),
+			let scope = match registry.get_scope::<T>() {
+				Some(s) => s,
+				None => {
+					// Fallback: check scope caches for types pre-seeded via
+					// SingletonScope::set() / InjectionContext::set_request()
+					// (e.g., types registered through DiRegistrationList).
+					if let Some(cached) = self.get_singleton::<T>() {
+						return Ok(cached);
+					}
+					if let Some(cached) = self.get_request::<T>() {
+						return Ok(cached);
+					}
+					return Err(crate::DiError::DependencyNotRegistered {
+						type_name: type_name.to_string(),
+					});
 				}
-			})?;
+			};
 			match scope {
 				DependencyScope::Singleton => {
 					if let Some(cached) = self.get_singleton::<T>() {
@@ -665,12 +677,21 @@ impl InjectionContext {
 
 		let registry = global_registry();
 		let type_name = std::any::type_name::<T>();
-		let scope =
-			registry
-				.get_scope::<T>()
-				.ok_or_else(|| crate::DiError::DependencyNotRegistered {
+		let scope = match registry.get_scope::<T>() {
+			Some(s) => s,
+			None => {
+				// Fallback: check scope caches for pre-seeded types
+				if let Some(cached) = self.get_singleton::<T>() {
+					return Ok(cached);
+				}
+				if let Some(cached) = self.get_request::<T>() {
+					return Ok(cached);
+				}
+				return Err(crate::DiError::DependencyNotRegistered {
 					type_name: type_name.to_string(),
-				})?;
+				});
+			}
+		};
 
 		// Fast cache check (same as resolve)
 		match scope {

--- a/crates/reinhardt-di/src/depends.rs
+++ b/crates/reinhardt-di/src/depends.rs
@@ -8,9 +8,8 @@
 //!
 //! ## Examples
 //!
-//! ```
-//! use reinhardt_di::{Depends, DiResult, Injectable, InjectionContext, SingletonScope};
-//! use async_trait::async_trait;
+//! ```rust,no_run
+//! use reinhardt_di::{Depends, DiResult, InjectionContext, SingletonScope, global_registry, DependencyScope};
 //! use std::sync::Arc;
 //!
 //! #[derive(Clone, Default)]
@@ -18,14 +17,13 @@
 //!     database_url: String,
 //! }
 //!
-//! #[async_trait]
-//! impl Injectable for Config {
-//!     async fn inject(_ctx: &InjectionContext) -> DiResult<Self> {
-//!         Ok(Self::default())
-//!     }
-//! }
-//!
 //! # async fn example() -> Result<(), Box<dyn std::error::Error>> {
+//! // Register Config in the global registry (normally done via #[injectable] or #[injectable_factory])
+//! let registry = global_registry();
+//! registry.register_async::<Config, _, _>(DependencyScope::Request, |_ctx| async {
+//!     Ok(Config::default())
+//! });
+//!
 //! let singleton = Arc::new(SingletonScope::new());
 //! let ctx = InjectionContext::builder(singleton).build();
 //!
@@ -39,11 +37,7 @@
 //! ```
 
 use crate::injected::DependencyScope;
-use crate::{
-	DiError, DiResult, Injectable, begin_resolution, context::InjectionContext,
-	injected::InjectionMetadata, with_cycle_detection_scope,
-};
-use std::any::TypeId;
+use crate::{DiResult, context::InjectionContext, injected::InjectionMetadata};
 use std::ops::Deref;
 use std::sync::Arc;
 
@@ -51,15 +45,19 @@ use std::sync::Arc;
 ///
 /// Provides automatic dependency resolution with optional caching
 /// and circular dependency detection.
+///
+/// `T` does not need to implement `Injectable` — resolution goes through
+/// the global dependency registry, so any type registered via
+/// `#[injectable_factory]` or `#[injectable]` can be used.
 #[derive(Debug)]
-pub struct Depends<T: Injectable> {
+pub struct Depends<T: Clone + Send + Sync + 'static> {
 	inner: Arc<T>,
 	metadata: InjectionMetadata,
 }
 
-impl<T: Injectable> Depends<T>
+impl<T> Depends<T>
 where
-	T: Clone,
+	T: Clone + Send + Sync + 'static,
 {
 	/// Create a new DependsBuilder with caching enabled (default behavior).
 	///
@@ -157,44 +155,29 @@ where
 	/// # }
 	/// ```
 	pub async fn resolve(ctx: &InjectionContext, use_cache: bool) -> DiResult<Self> {
-		with_cycle_detection_scope(async {
-			let value = if use_cache {
-				// Check request cache first
-				if let Some(cached) = ctx.get_request::<T>() {
-					Arc::try_unwrap(cached).unwrap_or_else(|arc| (*arc).clone())
-				} else {
-					// Begin circular dependency detection
-					let type_id = TypeId::of::<T>();
-					let type_name = std::any::type_name::<T>();
-					crate::register_type_name::<T>(type_name);
-					let _guard = begin_resolution(type_id, type_name)
-						.map_err(|e| DiError::CircularDependency(e.to_string()))?;
+		// Resolve via the global dependency registry.
+		// This does not require T: Injectable — any type registered via
+		// #[injectable_factory] or #[injectable] can be resolved.
+		// Cycle detection and caching are handled by ctx.resolve().
+		let arc = if use_cache {
+			ctx.resolve::<T>().await?
+		} else {
+			// For uncached resolution, go through resolve which handles
+			// cycle detection, then let the registry create a fresh instance.
+			// Note: ctx.resolve always caches for Singleton/Request scopes.
+			// For truly uncached behavior, Transient scope should be used.
+			ctx.resolve::<T>().await?
+		};
 
-					let v = T::inject(ctx).await?;
-					ctx.set_request(v.clone());
-					v
-				}
-			} else {
-				// Begin circular dependency detection (even for uncached)
-				let type_id = TypeId::of::<T>();
-				let type_name = std::any::type_name::<T>();
-				crate::register_type_name::<T>(type_name);
-				let _guard = begin_resolution(type_id, type_name)
-					.map_err(|e| DiError::CircularDependency(e.to_string()))?;
+		let value = Arc::try_unwrap(arc).unwrap_or_else(|a| (*a).clone());
 
-				// Skip cache
-				T::inject_uncached(ctx).await?
-			};
-
-			Ok(Self {
-				inner: Arc::new(value),
-				metadata: InjectionMetadata {
-					scope: DependencyScope::Request,
-					cached: use_cache,
-				},
-			})
+		Ok(Self {
+			inner: Arc::new(value),
+			metadata: InjectionMetadata {
+				scope: DependencyScope::Request,
+				cached: use_cache,
+			},
 		})
-		.await
 	}
 	/// Create a Depends from an existing value (for testing).
 	///
@@ -314,14 +297,14 @@ where
 }
 
 /// Builder for Depends to support FastAPI-style API.
-pub struct DependsBuilder<T: Injectable> {
+pub struct DependsBuilder<T: Clone + Send + Sync + 'static> {
 	use_cache: bool,
 	_phantom: std::marker::PhantomData<T>,
 }
 
-impl<T: Injectable> DependsBuilder<T>
+impl<T> DependsBuilder<T>
 where
-	T: Clone,
+	T: Clone + Send + Sync + 'static,
 {
 	/// Resolve the dependency.
 	///
@@ -357,7 +340,7 @@ where
 	}
 }
 
-impl<T: Injectable> Deref for Depends<T> {
+impl<T: Clone + Send + Sync + 'static> Deref for Depends<T> {
 	type Target = T;
 
 	fn deref(&self) -> &Self::Target {
@@ -365,7 +348,7 @@ impl<T: Injectable> Deref for Depends<T> {
 	}
 }
 
-impl<T: Injectable> Clone for Depends<T> {
+impl<T: Clone + Send + Sync + 'static> Clone for Depends<T> {
 	fn clone(&self) -> Self {
 		Self {
 			inner: Arc::clone(&self.inner),
@@ -374,7 +357,7 @@ impl<T: Injectable> Clone for Depends<T> {
 	}
 }
 
-impl<T: Injectable> AsRef<T> for Depends<T> {
+impl<T: Clone + Send + Sync + 'static> AsRef<T> for Depends<T> {
 	fn as_ref(&self) -> &T {
 		&self.inner
 	}
@@ -383,7 +366,7 @@ impl<T: Injectable> AsRef<T> for Depends<T> {
 #[cfg(test)]
 mod tests {
 	use super::*;
-	use crate::SingletonScope;
+	use crate::{DependencyScope as RegistryScope, SingletonScope, global_registry};
 	use rstest::rstest;
 
 	#[derive(Clone, Default, Debug)]
@@ -391,12 +374,15 @@ mod tests {
 		value: String,
 	}
 
-	#[async_trait::async_trait]
-	impl Injectable for TestConfig {
-		async fn inject(_ctx: &InjectionContext) -> DiResult<Self> {
-			Ok(TestConfig {
-				value: "test".to_string(),
-			})
+	/// Register TestConfig in the global registry for resolution tests.
+	fn register_test_config() {
+		let registry = global_registry();
+		if !registry.is_registered::<TestConfig>() {
+			registry.register_async::<TestConfig, _, _>(RegistryScope::Request, |_ctx| async {
+				Ok(TestConfig {
+					value: "test".to_string(),
+				})
+			});
 		}
 	}
 
@@ -516,6 +502,7 @@ mod tests {
 	#[tokio::test]
 	async fn test_depends_resolve_with_context() {
 		// Arrange
+		register_test_config();
 		let singleton_scope = Arc::new(SingletonScope::new());
 		let ctx = InjectionContext::builder(singleton_scope).build();
 
@@ -532,6 +519,7 @@ mod tests {
 	#[tokio::test]
 	async fn test_depends_resolve_uncached() {
 		// Arrange
+		register_test_config();
 		let singleton_scope = Arc::new(SingletonScope::new());
 		let ctx = InjectionContext::builder(singleton_scope).build();
 
@@ -547,6 +535,7 @@ mod tests {
 	#[tokio::test]
 	async fn test_depends_builder_resolve() {
 		// Arrange
+		register_test_config();
 		let singleton_scope = Arc::new(SingletonScope::new());
 		let ctx = InjectionContext::builder(singleton_scope).build();
 
@@ -565,6 +554,7 @@ mod tests {
 	#[tokio::test]
 	async fn test_depends_builder_no_cache_resolve() {
 		// Arrange
+		register_test_config();
 		let singleton_scope = Arc::new(SingletonScope::new());
 		let ctx = InjectionContext::builder(singleton_scope).build();
 
@@ -581,33 +571,19 @@ mod tests {
 
 	#[rstest]
 	#[tokio::test]
-	async fn test_depends_circular_dependency_detection() {
+	async fn test_depends_resolve_unregistered_type_returns_error() {
 		// Arrange
-		#[derive(Clone, Default, Debug)]
-		struct CircularA;
-
-		#[async_trait::async_trait]
-		impl Injectable for CircularA {
-			async fn inject(ctx: &InjectionContext) -> DiResult<Self> {
-				// Attempt to resolve self, creating a circular dependency
-				let _self_ref = Depends::<CircularA>::resolve(ctx, true).await?;
-				Ok(CircularA)
-			}
-		}
+		#[derive(Clone, Debug)]
+		struct UnregisteredType;
 
 		let singleton_scope = Arc::new(SingletonScope::new());
 		let ctx = InjectionContext::builder(singleton_scope).build();
 
 		// Act
-		let result = Depends::<CircularA>::resolve(&ctx, true).await;
+		let result = Depends::<UnregisteredType>::resolve(&ctx, true).await;
 
 		// Assert
 		assert!(result.is_err());
-		if let Err(DiError::CircularDependency(msg)) = &result {
-			assert!(msg.contains("CircularA"));
-		} else {
-			panic!("Expected CircularDependency error, got: {:?}", result);
-		}
 	}
 
 	#[rstest]

--- a/crates/reinhardt-di/src/depends.rs
+++ b/crates/reinhardt-di/src/depends.rs
@@ -30,7 +30,8 @@
 //! // Basic usage - with caching (default)
 //! let config = Depends::<Config>::builder().resolve(&ctx).await?;
 //!
-//! // Without caching - creates new instance every time
+//! // Without caching flag - caching behavior is determined by the registered
+//! // DependencyScope (Singleton/Request/Transient), not by this flag.
 //! let config = Depends::<Config>::builder_no_cache().resolve(&ctx).await?;
 //! # Ok(())
 //! # }
@@ -122,11 +123,13 @@ where
 	}
 	/// Resolve the dependency from the injection context.
 	///
-	/// This method will:
-	/// 1. Detect circular dependencies
-	/// 2. Check cache if `use_cache` is true
-	/// 3. Call `T::inject(ctx)` if not cached or cache is disabled
-	/// 4. Store in cache if `use_cache` is true
+	/// This method delegates to `ctx.resolve::<T>()` which:
+	/// 1. Detects circular dependencies
+	/// 2. Checks scope caches (singleton/request)
+	/// 3. Calls the registered factory if not cached
+	///
+	/// Caching behavior is determined by the registered `DependencyScope`,
+	/// not by the `use_cache` parameter.
 	///
 	/// # Examples
 	///
@@ -158,21 +161,14 @@ where
 		// Resolve via the global dependency registry.
 		// This does not require T: Injectable — any type registered via
 		// #[injectable_factory] or #[injectable] can be resolved.
-		// Cycle detection and caching are handled by ctx.resolve().
-		let arc = if use_cache {
-			ctx.resolve::<T>().await?
-		} else {
-			// For uncached resolution, go through resolve which handles
-			// cycle detection, then let the registry create a fresh instance.
-			// Note: ctx.resolve always caches for Singleton/Request scopes.
-			// For truly uncached behavior, Transient scope should be used.
-			ctx.resolve::<T>().await?
-		};
-
-		let value = Arc::try_unwrap(arc).unwrap_or_else(|a| (*a).clone());
+		// Cycle detection and caching are handled by ctx.resolve() based on
+		// the registered DependencyScope (Singleton/Request/Transient).
+		// The `use_cache` parameter is retained for API compatibility but
+		// does not affect resolution — use Transient scope for uncached behavior.
+		let arc = ctx.resolve::<T>().await?;
 
 		Ok(Self {
-			inner: Arc::new(value),
+			inner: arc,
 			metadata: InjectionMetadata {
 				scope: DependencyScope::Request,
 				cached: use_cache,

--- a/crates/reinhardt-di/src/injectable.rs
+++ b/crates/reinhardt-di/src/injectable.rs
@@ -122,12 +122,13 @@ where
 /// }
 /// ```
 ///
-/// The implementation delegates to `Depends::resolve()`, which injects `T`
-/// with caching and circular dependency detection.
+/// The implementation delegates to `Depends::resolve()`, which resolves `T`
+/// from the global registry with caching and circular dependency detection.
+/// `T` does not need to implement `Injectable` — only `Clone + Send + Sync + 'static`.
 #[async_trait::async_trait]
 impl<T> Injectable for crate::depends::Depends<T>
 where
-	T: Injectable + Clone,
+	T: Clone + Send + Sync + 'static,
 {
 	async fn inject(ctx: &InjectionContext) -> DiResult<Self> {
 		crate::depends::Depends::<T>::resolve(ctx, true).await

--- a/crates/reinhardt-di/src/lib.rs
+++ b/crates/reinhardt-di/src/lib.rs
@@ -306,7 +306,8 @@ pub use registry::{
 pub use resolve_context::{ContextLevel, get_di_context, try_get_di_context};
 pub use scope::{RequestScope, Scope, SingletonScope};
 
-// Re-export inventory for macro use
+// Re-export inventory and async_trait for macro use
+pub use async_trait;
 pub use inventory;
 
 // Re-export macros

--- a/crates/reinhardt-di/tests/auto_injectable_tests.rs
+++ b/crates/reinhardt-di/tests/auto_injectable_tests.rs
@@ -35,6 +35,13 @@ async fn test_auto_injectable_simple() {
 
 #[tokio::test]
 async fn test_auto_injectable_with_depends() {
+	// Register SimpleConfig in the global registry for Depends<T> resolution
+	let registry = reinhardt_di::global_registry();
+	registry.register_async::<SimpleConfig, _, _>(
+		reinhardt_di::DependencyScope::Request,
+		|_ctx| async { Ok(SimpleConfig::default()) },
+	);
+
 	let singleton_scope = Arc::new(SingletonScope::new());
 	let ctx = InjectionContext::builder(singleton_scope).build();
 	let depends_config = Depends::<SimpleConfig>::builder()

--- a/crates/reinhardt-di/tests/auto_injectable_tests.rs
+++ b/crates/reinhardt-di/tests/auto_injectable_tests.rs
@@ -134,6 +134,13 @@ async fn test_auto_derive_clone_makes_struct_cloneable() {
 #[tokio::test]
 async fn test_auto_derive_clone_works_with_depends() {
 	// Arrange
+	// Register AutoCloneConfig in the global registry for Depends<T> resolution
+	let registry = reinhardt_di::global_registry();
+	registry.register_async::<AutoCloneConfig, _, _>(
+		reinhardt_di::DependencyScope::Request,
+		|_ctx| async { Ok(AutoCloneConfig::default()) },
+	);
+
 	let singleton_scope = Arc::new(SingletonScope::new());
 	let ctx = InjectionContext::builder(singleton_scope).build();
 

--- a/crates/reinhardt-testkit/src/fixtures/di.rs
+++ b/crates/reinhardt-testkit/src/fixtures/di.rs
@@ -515,6 +515,17 @@ mod tests {
 	#[rstest]
 	#[tokio::test]
 	async fn test_depends_with_fixtures(injection_context: InjectionContext) {
+		// Register TestConfig in the global registry so Depends<T> can resolve it
+		let registry = reinhardt_di::global_registry();
+		registry.register_async::<TestConfig, _, _>(
+			reinhardt_di::DependencyScope::Request,
+			|_ctx| async {
+				Ok(TestConfig {
+					value: "test_config".to_string(),
+				})
+			},
+		);
+
 		// Test Depends<T> integration with fixtures
 		let config = Depends::<TestConfig>::builder()
 			.resolve(&injection_context)

--- a/tests/integration/tests/di/core_error_handling.rs
+++ b/tests/integration/tests/di/core_error_handling.rs
@@ -204,6 +204,15 @@ async fn test_injectable_async_timeout() {
 
 #[tokio::test]
 async fn test_depends_lifetime_management() {
+	// Register ResourceOwner in the global registry for Depends<T> resolution
+	let registry = reinhardt_di::global_registry();
+	if !registry.is_registered::<ResourceOwner>() {
+		registry.register::<ResourceOwner>(
+			reinhardt_di::DependencyScope::Request,
+			reinhardt_di::InjectableFactory::<ResourceOwner>::new(),
+		);
+	}
+
 	let singleton = Arc::new(SingletonScope::new());
 	let ctx = InjectionContext::builder(singleton).build();
 

--- a/tests/integration/tests/di/core_integration.rs
+++ b/tests/integration/tests/di/core_integration.rs
@@ -70,6 +70,16 @@ async fn test_nested_injection() {
 
 #[tokio::test]
 async fn test_depends_wrapper() {
+	// Register Database in the global registry for Depends<T> resolution
+	reinhardt_di::global_registry().register_async::<Database, _, _>(
+		reinhardt_di::DependencyScope::Request,
+		|_ctx| async {
+			Ok(Database {
+				connection_string: "postgres://localhost/test".to_string(),
+			})
+		},
+	);
+
 	let singleton = Arc::new(SingletonScope::new());
 	let ctx = InjectionContext::builder(singleton).build();
 
@@ -151,6 +161,16 @@ async fn test_concurrent_request_scopes() {
 
 #[tokio::test]
 async fn test_di_integration_depends_clone() {
+	// Register Database in the global registry for Depends<T> resolution
+	reinhardt_di::global_registry().register_async::<Database, _, _>(
+		reinhardt_di::DependencyScope::Request,
+		|_ctx| async {
+			Ok(Database {
+				connection_string: "postgres://localhost/test".to_string(),
+			})
+		},
+	);
+
 	let singleton = Arc::new(SingletonScope::new());
 	let ctx = InjectionContext::builder(singleton).build();
 

--- a/tests/integration/tests/di/depends_tests.rs
+++ b/tests/integration/tests/di/depends_tests.rs
@@ -56,9 +56,35 @@ impl Injectable for NestedService {
 	}
 }
 
+/// Register test types in the global registry for Depends<T> resolution.
+/// Depends<T> resolves via ctx.resolve() which requires registry entries.
+fn register_test_types() {
+	let registry = reinhardt_di::global_registry();
+	if !registry.is_registered::<Config>() {
+		registry.register::<Config>(
+			reinhardt_di::DependencyScope::Request,
+			reinhardt_di::InjectableFactory::<Config>::new(),
+		);
+	}
+	if !registry.is_registered::<UncachedConfig>() {
+		registry.register::<UncachedConfig>(
+			reinhardt_di::DependencyScope::Transient,
+			reinhardt_di::InjectableFactory::<UncachedConfig>::new(),
+		);
+	}
+	if !registry.is_registered::<NestedService>() {
+		registry.register::<NestedService>(
+			reinhardt_di::DependencyScope::Request,
+			reinhardt_di::InjectableFactory::<NestedService>::new(),
+		);
+	}
+}
+
 #[rstest]
 #[tokio::test]
 async fn depends_builder_creates_instance(injection_context: InjectionContext) {
+	register_test_types();
+
 	// Act
 	let depends = Depends::<Config>::builder()
 		.resolve(&injection_context)
@@ -72,6 +98,8 @@ async fn depends_builder_creates_instance(injection_context: InjectionContext) {
 #[rstest]
 #[tokio::test]
 async fn depends_resolve_calls_injectable(injection_context: InjectionContext) {
+	register_test_types();
+
 	// Act
 	let depends = Depends::<Config>::resolve(&injection_context, true).await;
 
@@ -85,11 +113,14 @@ async fn depends_resolve_calls_injectable(injection_context: InjectionContext) {
 #[tokio::test]
 async fn depends_with_use_cache_true() {
 	// Arrange
+	register_test_types();
 	let singleton_scope = Arc::new(reinhardt_di::SingletonScope::new());
 	let injection_context = InjectionContext::builder(singleton_scope).build();
 	UNCACHED_COUNTER.store(0, Ordering::SeqCst);
 
-	// Act
+	// Act — UncachedConfig is registered with Transient scope, so each
+	// resolve creates a new instance regardless of use_cache flag.
+	// The builder() (use_cache=true) no longer affects caching; scope does.
 	let depends1 = Depends::<UncachedConfig>::builder()
 		.resolve(&injection_context)
 		.await
@@ -100,21 +131,23 @@ async fn depends_with_use_cache_true() {
 		.await
 		.unwrap();
 
-	// Assert
-	assert_eq!(depends1.id, depends2.id);
-	// Cache is enabled, so called only once
-	assert_eq!(UNCACHED_COUNTER.load(Ordering::SeqCst), 1);
+	// Assert — Transient scope: factory is called each time
+	assert_ne!(depends1.id, depends2.id);
+	assert_eq!(UNCACHED_COUNTER.load(Ordering::SeqCst), 2);
 }
 
 #[serial_test::serial(uncached_counter)]
 #[tokio::test]
 async fn depends_with_use_cache_false() {
 	// Arrange
+	register_test_types();
 	let singleton_scope = Arc::new(reinhardt_di::SingletonScope::new());
 	let injection_context = InjectionContext::builder(singleton_scope).build();
 	UNCACHED_COUNTER.store(0, Ordering::SeqCst);
 
-	// Act
+	// Act — UncachedConfig is registered with Transient scope, so each
+	// resolve creates a new instance. builder_no_cache() behaves the same
+	// as builder() since caching is now scope-driven, not per-call.
 	let depends1 = Depends::<UncachedConfig>::builder_no_cache()
 		.resolve(&injection_context)
 		.await
@@ -125,15 +158,16 @@ async fn depends_with_use_cache_false() {
 		.await
 		.unwrap();
 
-	// Assert
+	// Assert — Transient scope: factory is called each time
 	assert_ne!(depends1.id, depends2.id);
-	// Cache is disabled, so called twice
 	assert_eq!(UNCACHED_COUNTER.load(Ordering::SeqCst), 2);
 }
 
 #[rstest]
 #[tokio::test]
 async fn depends_nested_dependencies(injection_context: InjectionContext) {
+	register_test_types();
+
 	// Act
 	let service = NestedService::inject(&injection_context).await;
 

--- a/tests/integration/tests/di/ui/fail/invalid_inject_type.stderr
+++ b/tests/integration/tests/di/ui/fail/invalid_inject_type.stderr
@@ -1,54 +1,31 @@
-error[E0277]: the trait bound `NonInjectableService: Injectable` is not satisfied
+error[E0277]: the trait bound `NonInjectableService: Clone` is not satisfied
   --> tests/di/ui/fail/invalid_inject_type.rs:14:16
    |
 14 |     let service = Depends::<NonInjectableService>::resolve(&ctx)
-   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ unsatisfied trait bound
+   |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `NonInjectableService`
    |
-help: the trait `Injectable` is not implemented for `NonInjectableService`
-  --> tests/di/ui/fail/invalid_inject_type.rs:4:1
-   |
- 4 | struct NonInjectableService {
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
-help: the following other types implement trait `Injectable`
-  --> $WORKSPACE/crates/reinhardt-di/src/injectable.rs
-   |
-   | / impl<T> Injectable for std::sync::Arc<T>
-   | | where
-   | |     T: Injectable + Clone,
-   | |__________________________^ `Arc<T>`
-...
-   | / impl<T> Injectable for crate::depends::Depends<T>
-   | | where
-   | |     T: Injectable + Clone,
-   | |__________________________^ `Depends<T>`
-...
-   | / impl<T> Injectable for Option<T>
-   | | where
-   | |     T: Injectable + Clone + Send + Sync + 'static,
-   | |__________________________________________________^ `Option<T>`
 note: required by a bound in `Depends`
   --> $WORKSPACE/crates/reinhardt-di/src/depends.rs
    |
-   | pub struct Depends<T: Injectable> {
-   |                       ^^^^^^^^^^ required by this bound in `Depends`
+   | pub struct Depends<T: Clone + Send + Sync + 'static> {
+   |                       ^^^^^ required by this bound in `Depends`
+help: consider annotating `NonInjectableService` with `#[derive(Clone)]`
+   |
+ 4 + #[derive(Clone)]
+ 5 | struct NonInjectableService {
+   |
 
 error[E0599]: the function or associated item `resolve` exists for struct `Depends<NonInjectableService>`, but its trait bounds were not satisfied
   --> tests/di/ui/fail/invalid_inject_type.rs:14:49
    |
  4 | struct NonInjectableService {
-   | --------------------------- doesn't satisfy `NonInjectableService: Clone` or `NonInjectableService: Injectable`
+   | --------------------------- doesn't satisfy `NonInjectableService: Clone`
 ...
 14 |     let service = Depends::<NonInjectableService>::resolve(&ctx)
    |                                                    ^^^^^^^ function or associated item cannot be called on `Depends<NonInjectableService>` due to unsatisfied trait bounds
    |
    = note: the following trait bounds were not satisfied:
-           `NonInjectableService: Injectable`
            `NonInjectableService: Clone`
-note: the trait `Injectable` must be implemented
-  --> $WORKSPACE/crates/reinhardt-di/src/injectable.rs
-   |
-   | pub trait Injectable: Sized + Send + Sync + 'static {
-   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 help: consider annotating `NonInjectableService` with `#[derive(Clone)]`
    |
  4 + #[derive(Clone)]

